cpp/include/Ice/SSL/ConnectionInfoF.h

```diff
@@ -5,12 +5,22 @@
 #ifndef ICE_SSL_CONNECTION_INFO_F_H
 #define ICE_SSL_CONNECTION_INFO_F_H
 
+#include "Config.h"
+
 #include <memory>
 
 namespace Ice::SSL
 {
-    class ConnectionInfo;
-    using ConnectionInfoPtr = std::shared_ptr<ConnectionInfo>;
+#if defined(ICE_USE_SCHANNEL)
+    class SchannelConnectionInfo;
```

Comment:
I would include the aliasing here (using ConnectionInfo = SchannelConnectionInfo) and remove it from ConnectionInfo.h